udata_recommendations/commands.py

```diff
@@ -11,36 +14,56 @@
 
 log = logging.getLogger(__name__)
 
+SCHEMA_FILENAME = 'schema.json'
+
 
 @cli.group()
 def recommendations():
     '''Recommendations related operations'''
     pass
 
 
-def clean_datasets_recommendations():
-    log.info('Cleaning up dataset recommendations...')
+def clean_datasets_recommendations(source):
+    log.info(f'Cleaning up dataset recommendations from source {source}')
+
     datasets = Dataset.objects.filter(**{
-        'extras__recommendations__exists': True,
-    }).update(**{
-        'unset__extras__recommendations': True,
+        'extras__recommendations:sources__contains': source,
     })
-    success('Cleaned up %s dataset(s).' % datasets)
 
+    for dataset in datasets:
+        new_sources = set(
+            [s for s in dataset.extras['recommendations:sources'] if s != source]
+        )
+
+        # Dataset had only this source: delete all recommendations
+        if len(new_sources) == 0:
+            dataset.update(**{
+                'unset__extras__recommendations': True,
+                'unset__extras__recommendations:sources': True,
+            })
+            continue
+
+        # Dataset had at least one other source, keep only other sources
+        new_recommendations = [
+            r for r in dataset.extras['recommendations']
+            if r['source'] != source
+        ]
+        dataset.extras['recommendations:sources'] = list(new_sources)
+        dataset.extras['recommendations'] = new_recommendations
+        dataset.save()
+
+    success(f"Cleaned up {len(datasets)} dataset(s)")
+
+
+def get_recommendations_data(url):
+    response = requests.get(url, timeout=10)
+    response.raise_for_status()
+    data = response.json()
+
+    with open(Path(__file__).parent / SCHEMA_FILENAME) as f:
```

Comment:
There's a pythonic way to do this: https://docs.python.org/3/distutils/setupscript.html#installing-package-data

Yours might run into compatibility issues on some platforms / setups (aka _ce serait trop simple_).